# BEE313 Problem 1.1

## Objective 
The goal of this problem was to determine the volumetric water conent when a new wine grape would become stressed in Sandy Soil. 
## Methods:
The campbell pedotransfer function was used in order to determine the pressure head associated with wine grapes becoming stressed in silty-clay soils. Then using this same pressure head, the parameters for sandy soil were applied to the pedo transfer function and it was solved for the volumetric content of the sandy soil.


ph_silt = air_ph_1*(porosity_1/X1)**b_1

ph_sand = air_ph_2*(porosity_2/X2)**b_2
X2 = 1/(((ph_sand/air_ph)**(1/b_2))*(1/(porosity)))


In [15]:
# X1 is known volumetric content...
# X2 is unknown volumetric content
# ph_silt = ph_sand... assume that... then solbe for X2

# First solved for the pressure head associated wtih the soil type of silty clay and the volumetric soil water content of .42 cm3/cm3
X1 = .42
porosity_1 = .492
air_ph_1 = 49
b_1 = 10.4

ph_silt = air_ph_1*(porosity_1/X1)**b_1

print(f"pressure head {ph_silt:.3f}")

# Then solved for the volemtric water content in the sand soil assuming same pressure head...
porosity_2 = .395
air_ph_2 = 12.1
ph_sand = ph_silt
b_2 = 4.05
#X2 = 1/(((ph_sand/air_ph_2)**(1/b_2))*(1/(porosity_2)))
X2 = (air_ph_2*((porosity_2**b_2)/ph_sand))**(1/b_2)
print(" ")
print("Volumetric Water Content of Sand")
print(f"{X2:.3f}")

pressure head 254.004
 
Volumetric Water Content of Sand
0.186


## Results:
The volumetric water conent of the sandy soil was determined to be .186cm^3/cm^3 at a presure head of 254 cm. 
## Discussion:

The pressure head for these two soils was deemed to be the same, and due to the fact that the two soils have different pore sizes, this allows for different rates of evapotranspiration and drainage. This ultimately leads to less water storage within the larger pores, and thus thus indicating that there would be a lower volumetric soil water content due to this reduction in storage capacity. 


# BEE313 Problem 1.2

## Objective 
The goal of this problem was to determine the amount of surface runoff that occurred, after the water had infiltrated 10cm down to the ends soil moisture probes. It was assumed that the time to pond was not the same as the time that the water reached teh soil moisture probes.  
## Methods:
The time to pond was first determined. Then after this point the cumulative infiltration that occurred up until this point of ponding was found. Then the cumulative precipitation was calculated for when the wetting front reached the soil moisture probes 10cm down. With this value and the time to pond infiltration the time that it took to infiltrate to the probes was found. Finally the excess infiltration was found by multiplying the time by the precipitation rate and subtracting the cumultative infiltration over that period from that.



In [16]:
# Infiltration Excess ... saturation from above
# So this means all pores are fill...

import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
# CLAY SOIL

# KNOWNS:
k_h_sec = 1.28*(10**-4)  # cm/s
k_h_hr = k_h_sec * 60*60  # cm/hr
psi_ae = 40.5  # cm
w = 40/10  # cm/hour
z_f = 10  # cm
b = 11.4
time_total = 1  # hour
# porosity of clay...
porosity = .482  # proportion...
# In Arid Region -> Given the inital water content as a proportion of volume...
theta_init = .35  # proportion.

# found psi_f...
psi_f = ((2*b + 3)/(2*b + 6))*psi_ae

# Calculated the Time to Pond.
###### w > K_H
# Then when we get time to pond
theta_w = porosity
## Time to pond...
Tp = (k_h_hr * psi_f*(theta_w - theta_init))/(w*(w-k_h_hr))


# Amount of infiltration that occurs up until point of ponding..
## Assumed that infiltration is constant at rainfall rate of 4cm/hour
F_Tp = w * Tp 


# Then calculated the cumulative infiltration at the depth of... soil probes
F_T = z_f*(porosity-theta_init)
# Time: That it takes to reach F_Tp
t = ((F_T - F_Tp)/k_h_hr)+((psi_f*(porosity-theta_init)/k_h_hr))*np.log((F_Tp+psi_f*(porosity-theta_init))/(F_T+psi_f*(porosity-theta_init)))+Tp

# Infiltration excess
# Rainfall fall rate times time in hours t...
# F_T is the amou t of infiltration that has occurred
fexcess_T = w*t- F_T # rainfall - infiltration ponding... 
print(f"Surface Runoff: {fexcess_T:.3f} cm")
print(f"time to reach 10cm in soil: {t*60:.3f} minutes")

Surface Runoff: 0.317 cm
time to reach 10cm in soil: 24.558 minutes


## Resuts:
The surface runoff was determined to be .317cm.

## Discussion:
It took 25 minutes assuming that the soil was iniitally saturated for the water to infiltrate through the column and reach the soil moisture probes. After the ponding time was reached water was still able to infiltrate through the soil column, and is why the wetting front continued to move further through the soil profile.

# BEE313 Problem 1.3

## Objective 
The objective of this problem was to determine the height of the water table at Millersburg above the confining layer that is assocaited with the Willamette Valley. 
## Methods:
The Dupuit approximation was utilized to determine this height. The heights of the willamete and santiam surface elevations from sea level were accounted for by subtracted the height of the confining layer from them. This allowed the region above the confining layer to be computed to determine how much would correspond to the water table. The appropriate values were plugged into the equation and the units were converted such that they were in meters.

In [17]:
# What is the expected height of the water table?
# Confining layer of height of 45 , above sea level which is below the willamette valley
# Average Annual reachagre is 500 mm/ year, hydraulic conductivity is 51.8 m/day...
R_in = (500/(1000*365))  # m/year
k_h = 51.8  # m/day
h_o = 52 -45 # m
hL = 65 -45 # m
L = 6.2 * 1000  # m
x = 2.3 * 1000  # m

h_x = (((-R_in/k_h)*x**2)+(((hL**2)-(h_o**2))/L)*x+(R_in/k_h)*L*x + h_o**2)**(1/2)

print(f"Height of Water Table Above the Confining Layer {h_x:.3f} m")  # above the confined layer...

Height of Water Table Above the Confining Layer 20.406 m


## Results
The height of the water table above the confining layer was found to be 20.406m.

## Discussion
It is important to noted that in the computation of this value the heights above the confining layer needed to be accounted for since th hydraulic conductivity corresponds to the area above this layer. This was an important distinction to make in order to accurately compute a value for the height of the water table. If the heights from the sea level were computed initially then this water table height would be far larger. 

# BEE313 Problem 1.4

## Objective 

The goal of this problem was to determine the hydraulic conductivity of the confined aquifer. 
## Methods:
The Thiem equation was utilized. The Well was set to be the first points with the height being the thickness subtracted from the drawdown. The Monitoring point was the second point with the height being the thickness subtracted from the drawdown at that point as well. The discharge was converted to be cm^3/s. The kH was solved for by manipulating the equation. 

In [18]:
# monitoring well means that there is no pumping...

import numpy as np

Qpump = 500*100*100*100/(24*60*60)  # cm3/second

# Aquifer Thickness
H = 50*100

# Well (1)
h1 = H-(2*100)  # cm
r1 = .5*100  # cm

# monitoring point (2)
r2 = 30*100  # cm
h2 = H - (.25*100)  # cm

# Need to Solve for Hydraulic Conductivity

k_H = (Qpump/(2*(np.pi)*H*((h1 - h2))))*np.log(r1/r2)
print(f"Hydraulic Conductivity {k_H:.3f} cm/s ")

Hydraulic Conductivity 0.004 cm/s 


## Results
The hydraulic conductivity was found to be .004 cm/s.

## Discussion
This value makes sense for hydraulic conductivity to be low since water is less likely able to infiltrate easily into the aquifer. Hydraulic conductivity is directly proportional to drainage rates so this small value for hydraulic conductivity makes sense. 